addons/project_timesheet_holidays/tests/test_timesheet_holidays.py

```diff
@@ -127,3 +127,40 @@ def test_validate_without_timesheet(self):
         })
         holiday.with_user(SUPERUSER_ID).action_validate()
         self.assertEqual(len(holiday.timesheet_ids), 0, 'Number of generated timesheets should be zero since the leave type does not generate timesheet')
+
+    def test_timesheet_time_off_including_public_holiday(self):
+        """ Public holiday should not generate an entry if the employee is already on time off
```

Comment:
You don't really check this since you create the public holiday before creating the time off. So normally, one timesheet is generated and 4 timesheets will be generated for the time  off. 🤔 
```suggestion
        """ Generate one timesheet for the public holiday and 4 timesheets for the time off.
```

However, if you want to check if the timesheet is not generated for the public holiday, you should create the public holiday after creating the time off.

Reply:
I think the comment for the test is not correct, since you create first the public holiday, the timesheet for this public holiday will be generated and then only 3 timesheets will be generated for the other leaves taken by the employee.

Don't you agree with me? 🤔

Reply:
@xavierbol 👍